src/main/java/echoserver/EchoServer.java

```diff
@@ -19,15 +19,19 @@ public EchoServer(Listenable listener, Executor executor, Echoable echoer, Logga
     public void start() throws Exception {
         Connection socketConnection;
         while((socketConnection = this.listener.listen()) != null) {
-            Connection finalSocketConnection = socketConnection;
-            Runnable runnable = () -> {
-                try {
-                    this.echoer.echo(finalSocketConnection);
-                } catch (IOException e) {
-                    this.logger.log("Failed to echo connection.");
-                }
-            };
+            Runnable runnable = runnable(socketConnection);
             executor.execute(runnable);
         }
     }
+
+    private Runnable runnable(Connection connection) {
+        Runnable runnable = () -> {
+            try (Connection socketConnection = connection) {
+                this.echoer.echo(connection);
```

Comment:
even though `socketConnection` and `connection` are the same object, I would use `socketConnection` within the try for consistency.